### PR TITLE
Add support for Salesforce Lightning Components

### PR DESCRIPTION
**Pull request by davidsbond**
Added language support for Salesforce Lightning Components

## Description
Salesforce Lightning Components use the `.cmp`, `.auradoc`, `.design` & `.evt` extensions which are currently not supported, while Salesforce's proprietary language Apex is. 

## Checklist:
- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      -  [Many of these Apex based repositories use lightning components](https://github.com/search?utf8=%E2%9C%93&q=language%3AApex&type=Repositories&ref=advsearch&l=Apex&l=)
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
       - [Cumulus](https://github.com/SalesforceFoundation/Cumulus/tree/master/src/aura/HH_AddressMgr)
    - Sample license(s):
       - BSD-3-Clause
  - [x] I have included a syntax highlighting grammar.
  - [x] I have included a change to the heuristics to distinguish my language from others using the same extension.

### CODE REVIEW

**Comment by davidsbond** [comment]
Apologies for the number of commits for this, for some reason git was saying I had changed things that I hadn't. Should be resolved now

**Comment by lildude** [comment]
> Apologies for the number of commits for this, for some reason git was saying I had changed things that I hadn't. Should be resolved now

No probs, we squash all commits when we merge, so go nuts 😉

**Comment by davidsbond** [comment]
Is there an easy way to pick a colour that isn't too close to another?

**Comment by davidsbond** [comment]
Found a colour 👍

File: lib/linguist/heuristics.rb [hunk 1]
```diff
@@ -105,6 +105,12 @@ def call(data)
       end
     end
 
+    disambiguate ".auradoc" do |data|
```
Comment (pchaigno) [inline lib/linguist/heuristics.rb]: Heuristic rules are only required for conflicting languages. You can check in `languages.yml` and remove heuristic rules for extensions that are only used by Lightning Component.

File: lib/linguist/languages.yml [hunk 2]
```diff
@@ -2356,6 +2356,19 @@ Lex:
...
+  ace_mode: html
+  codemirror_mode: htmlmixed
+  codemirror_mime_type: text/html
+  extensions:
+    - ".auradoc"
+    - ".cmp"
+    - ".design"
+    - ".evt"
+  language_id: 432
+  tm_scope: none
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: You've entered `html` for the `ace_mode`. Couldn't HTML also be used for the `tm_scope` (grammar for source highlighting)?

File: lib/linguist/languages.yml [hunk 3]
```diff
@@ -5324,4 +5337,4 @@ xBase:
   - ".prw"
   tm_scope: source.harbour
   ace_mode: text
-  language_id: 421
+  language_id: 421
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: Please don't remove the newline at the end of the file.

**Comment by davidsbond** [comment]
Haven't had a lot of time to look back at this, will reopen when I can make the changes.

**Pull request closed without merging**